### PR TITLE
fix `_multiclass_stat_scores_update` in classification

### LINKED ISSUE #3068: Multiclass accuracy with micro and top-k does not work as expected.
## 🐛 Bug

I expect that `top_k` should impact accuracy, even in micro mode -- namely, in that `top_k` should mean that you get a prediction "correct" if the "right" logit is among the "top `top_k`" scores, then you average the number of corrects as per usual. This is not the behavior I see with torchmetrics now. Instead, I see this:

```python
>>> logits
tensor([[[0.0000, 0.1000, 0.5000, 0.4000],
         [0.0000, 0.2000, 0.7000, 0.1000]],

        [[0.0000, 0.4000, 0.3000, 0.3000],
         [1.0000, 0.0000, 0.0000, 0.0000]]])
>>> code
tensor([[3, 2],
        [1, 0]])
>>> logits.shape
torch.Size([2, 2, 4])
>>> code.shape
torch.Size([2, 2])
>>> acc = Accuracy(task="multiclass", ignore_index=0, num_classes=4, multidim_average="global", average="micro", top_k=4)
>>> acc(logits.transpose(2, 1), code)
tensor(0.6667)
>>> acc = Accuracy(task="multiclass", ignore_index=0, num_classes=4, multidim_average="global", average="micro", top_k=3)
>>> acc(logits.transpose(2, 1), code)
tensor(0.6667)
>>> acc = Accuracy(task="multiclass", ignore_index=0, num_classes=4, multidim_average="global", average="micro", top_k=2)
>>> acc(logits.transpose(2, 1), code)
tensor(0.6667)
>>> acc = Accuracy(task="multiclass", ignore_index=0, num_classes=4, multidim_average="global", average="micro", top_k=1)
>>> acc(logits.transpose(2, 1), code)
tensor(0.6667)
```

### Expected behavior

I would expect to see all accuracies with top_k >= 2 to be 1.0.

### Additional context

This line https://github.com/Lightning-AI/torchmetrics/blob/master/src/torchmetrics/functional/classification/accuracy.py#L85 ignores `top_k` in the micro case.

I think the fix to this issue may be related: https://github.com/Lightning-AI/torchmetrics/issues/3037

### PR DESCRIPTION
**Pull request by ved1beta**
## What does this PR do?

Fixes #3068

This PR fixes an issue with the multiclass accuracy calculation when using `top_k > 1` with `average="micro"`. The bug caused incorrect accuracy calculations in scenarios where predictions were provided as logits/probabilities and the correct class needed to be identified among the top-k predictions.

The fix ensures that:
1. The one-hot encoding path is always used when `top_k > 1`, regardless of the averaging method
2. A special case is added in `multiclass_accuracy` to properly handle top-k with micro averaging
3. Top-k selection is consistently applied across all evaluation scenarios

The PR includes test cases that demonstrate the issue and verify the fix works correctly. These tests show the alignment between manual calculations of top-k accuracy and the results from the metric implementation.

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--3078.org.readthedocs.build/en/3078/

fixes #3068

### CODE REVIEW

**Comment by rittik9** [comment]
Hi @Borda, this issue is already resolved by https://github.com/Lightning-AI/torchmetrics/pull/3117.
